diff --git a/dlls/user32/clipboard.c b/dlls/user32/clipboard.c
index a5138d66b2c..74cdd6aac7c 100644
--- a/dlls/user32/clipboard.c
+++ b/dlls/user32/clipboard.c
@@ -68,17 +68,17 @@ static int send_android_clipboard_data(android_clipboard_data_t clipboard_data)
 	net_requestcode = htonl(clipboard_data.request_code);
 	net_data_format = htonl(clipboard_data.format);
 	net_data_size = htonl(clipboard_data.size);
-	ret = send(sock_fd, &net_requestcode, sizeof(net_requestcode), 0);
+	ret = send(sock_fd, (const char *)&net_requestcode, sizeof(net_requestcode), 0);
 	if (ret == SOCKET_ERROR) {
 		TRACE( "Failed to send request code\n" );
 		return 0;
 	}
-	ret = send(sock_fd, &net_data_format, sizeof(net_data_format), 0);
+	ret = send(sock_fd, (const char *)&net_data_format, sizeof(net_data_format), 0);
 	if (ret == SOCKET_ERROR) {
 		TRACE( "Failed to send Windows clipboard data format\n" );
 		return 0;
 	}
-	ret = send(sock_fd, &net_data_size, sizeof(net_data_size), 0);
+	ret = send(sock_fd, (const char *)&net_data_size, sizeof(net_data_size), 0);
 	if (ret == SOCKET_ERROR) {
 		TRACE( "Failed to send Windows clipboard data size\n" );
 		return 0;
@@ -129,15 +129,15 @@ static android_clipboard_data_t receive_android_clipboard_data() {
 	clipboard_data.request_code = 3;
 	net_requestcode = htonl(clipboard_data.request_code);
 	
-	ret = send(sock_fd, &net_requestcode, sizeof(net_requestcode), 0);
+	ret = send(sock_fd, (const char *)&net_requestcode, sizeof(net_requestcode), 0);
 	if (ret == SOCKET_ERROR) {
 		TRACE( "Failed to send request code\n" );
 	}
-    ret = recv(sock_fd, &net_data_format, sizeof(net_data_format), 0);
+    ret = recv(sock_fd, (char *)&net_data_format, sizeof(net_data_format), 0);
     if (ret == SOCKET_ERROR) {
     	TRACE( "Failed to receive Android clipboard data format\n" );
     }
-    ret = recv(sock_fd, &net_data_size, sizeof(net_data_size), 0);
+    ret = recv(sock_fd, (char *)&net_data_size, sizeof(net_data_size), 0);
     if (ret == SOCKET_ERROR) {
     	TRACE( "Failed to receive Android clipboard data size\n" );
     }
