diff --git a/programs/winebrowser/main.c b/programs/winebrowser/main.c
index 94988b9b530..90a33baac2e 100644
--- a/programs/winebrowser/main.c
+++ b/programs/winebrowser/main.c
@@ -85,12 +85,12 @@ static int send_android_message(android_message_t message) {
 	}
 	int net_requestcode = htonl(message.request_code);
 	int net_data_length = htonl(strlen(message.data));
-	ret = send(sock_fd, &net_requestcode, sizeof(net_requestcode), 0);
+	ret = send(sock_fd, (const char *)&net_requestcode, sizeof(net_requestcode), 0);
 	if (ret == SOCKET_ERROR) {
 		TRACE( "Failed to send request code to server\n" );
 		return 0;
 	}
-	ret = send(sock_fd, &net_data_length, sizeof(net_data_length), 0);
+	ret = send(sock_fd, (const char *)&net_data_length, sizeof(net_data_length), 0);
 	if (ret == SOCKET_ERROR) {
 		TRACE( "Failed to send data length to server\n" );
 		return 0;
