From 3f2d650e7b0e6d81b88e1e175582be389a20e99b Mon Sep 17 00:00:00 2001
From: PabloLabs36 <pablolabs36@gmail.com>
Date: Fri, 27 Dec 2024 21:57:27 -0300
Subject: [PATCH 04/18] loader: Disable Light Theme by Default

---
 loader/wine.inf.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index cedcf16fba7..2fee4bded77 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -2419,7 +2419,6 @@ ErrorControl=1
 
 [ThemeManager]
 HKCU,"Software\Microsoft\Windows\CurrentVersion\ThemeManager","ThemeActive",2,"1"
-HKCU,"Software\Microsoft\Windows\CurrentVersion\ThemeManager","DllName",2,"%10%\resources\themes\light\light.msstyles"
 HKCU,"Software\Microsoft\Windows\CurrentVersion\ThemeManager","ColorName",2,"Blue"
 HKCU,"Software\Microsoft\Windows\CurrentVersion\ThemeManager","SizeName",2,"NormalSize"
 
-- 
2.55.0.windows.3

