diff --git a/dlls/ntdll/unix/env.c b/dlls/ntdll/unix/env.c
index cef5a5b679f..13cb1da9ba5 100644
--- a/dlls/ntdll/unix/env.c
+++ b/dlls/ntdll/unix/env.c
@@ -808,7 +808,19 @@ static void init_locale(void)
     const NLS_LOCALE_HEADER *locale_table;
     const NLS_LOCALE_DATA *locale;
     char *p;
+#ifdef __ANDROID__
+    all = getenv( "LC_ALL" );
+    if (!all)
+        all = "C.UTF-8";
 
+    if (!unix_to_win_locale( all, system_locale )) system_locale[0] = 0;
+    TRACE_(nls)( "Unix LC_ALL is %s, setting system locale to %s\n", debugstr_a(all), debugstr_a(system_locale) );
+
+    if (!unix_to_win_locale( all, user_locale )) user_locale[0] = 0;
+    TRACE_(nls)( "Unix LC_ALL is %s, user system locale to %s\n", debugstr_a(all), debugstr_a(user_locale) );
+#else
+    if (!unix_to_win_locale ( all, user_locale )) user_locale[0] = 0;     
+    
     if (!(all = setlocale( LC_ALL, "" )) && (all = getenv( "LC_ALL" )))
         FIXME_(nls)( "Failed to set LC_ALL to %s, is the locale supported?\n", debugstr_a(all) );
     if (!(ctype = setlocale( LC_CTYPE, "" )) && (ctype = getenv( "LC_CTYPE" )))
@@ -827,7 +839,7 @@ static void init_locale(void)
 
     if (!unix_to_win_locale( messages, user_locale )) user_locale[0] = 0;
     TRACE_(nls)( "Unix LC_MESSAGES is %s, user system locale to %s\n", debugstr_a(messages), debugstr_a(user_locale) );
-
+#endif
 #ifdef __APPLE__
     if (!system_locale[0])
     {
