diff --git a/bin/install_megadrivers.py b/bin/install_megadrivers.py
--- a/bin/install_megadrivers.py
+++ b/bin/install_megadrivers.py
@@ -24,9 +24,22 @@

 import argparse
 import os
+import shutil
 import sys


+def link(src, dest):
+    shutil.copyfile(src, dest)
+
+
+def unlink(src):
+    os.remove(src)
+
+
+os.link = link
+os.unlink = unlink
+
+
 def resolve_libdir(libdir):
     if os.path.isabs(libdir):
         destdir = os.environ.get('DESTDIR')
